machine/models/baseModel.py

```diff
@@ -0,0 +1,49 @@
+import torch.nn as nn
+import torch.nn.functional as F
+
+class Model(nn.Module):
+    """ 
+    Abstract base class for models.
+
+    Args:
+        encoder_module (baseRNN) :  module that encodes inputs
+        decoder_module (baseRNN, optional):   module to decode encoded inputs
+        decode_function (func, optional): function to generate symbols from output hidden states (default: F.log_softmax)
+
+    """
+
+    def __init__(self, encoder_module, decoder_module=None, decode_function=F.log_softmax):
+        super(Model, self).__init__()
+        self.encoder_module = encoder_module
+        self.decoder_module = decoder_module
+        self.decode_function = decode_function
+
+    def flatten_parameters(self):
+        self.encoder_module.rnn.flatten_parameters()
+        self.decoder_module.rnn.flatten_parameters()
+
```

Comment:
We need to add `reset_parameters` in the base. Either make it abstract or if you like the idea of making most of the baseclass work (like for `flatten_parametets`) then use something similar to : https://github.com/YannDubs/machine/blob/extrapolator/seq2seq/util/base.py#L84 (I store n_training_calls because I use it to update stuff, not sure other would need that although we could as it only stores one number). `weights_init` is a SOTA initialisation function for diffferent modules that I've defined here: https://github.com/YannDubs/machine/blob/extrapolator/seq2seq/util/initialization.py#L143

Reply:
Good point, I added it now as a method that needs to be implemented). It is non-abstract atm because it would make seq2seq unusable because it is not implemented there.